1. Провести дисперсионный анализ для определения того, есть ли различия среднего роста среди взрослых футболистов, хоккеистов и штангистов. Даны значения роста в трех группах случайно выбранных спортсменов:

Футболисты: 173, 175, 180, 178, 177, 185, 183, 182.
Хоккеисты: 177, 179, 180, 188, 177, 172, 171, 184, 180.
Штангисты: 172, 173, 169, 177, 166, 180, 178, 177, 172, 166, 170.

In [1]:
import numpy as np
from scipy import stats

In [20]:
k = 3

y1 = np.array([173, 175, 180, 178, 177, 185, 183, 182])
y2 = np.array([177, 179, 180, 188, 177, 172, 171, 184, 180])
y3 = np.array([172, 173, 169, 177, 166, 180, 178, 177, 172, 166, 170])

n1 = len(y1)
n2 = len(y2)
n3 = len(y3)
n = n1 + n2 + n3

print(f'Объем выборок: футболисты - {n1}, хоккеисты - {n2}, штангисты - {n3}. Общий объем - {n}.')

Объем выборок: футболисты - 8, хоккеисты - 9, штангисты - 11. Общий объем - 28.


In [13]:
y_1 = y1.mean()
y_2 = y2.mean()
y_3 = y3.mean()


y = np.hstack([y1, y2, y3])
y_ = y.mean()

In [14]:
S = 0
for i in y:
    S += (i - y_)**2
S

830.9642857142854

In [15]:
S_F = (y_1 - y_)**2 * n1 + (y_2 - y_)**2 * n2 + (y_3 - y_)**2 * n3
S_F

253.9074675324678

In [18]:
S_o = 0
for i in y1:
    S_o += (i - y_1)**2
for i in y2:
    S_o += (i - y_2)**2
for i in y3:
    S_o += (i - y_3)**2
S_o

577.0568181818181

In [19]:
S_F + S_o

830.9642857142859

In [21]:
sigma_F = S_F / (k - 1)
sigma_o = S_o / (n - k)

F = sigma_F / sigma_o
F

5.500053450812599

$F_c = 3.59$

$5.50 > 3.59$

Следовательно, различие между группами статистически значимое для $\alpha = 0.05$.

In [22]:
stats.f_oneway(y1, y2, y3)

F_onewayResult(statistic=5.500053450812596, pvalue=0.010482206918698694)

$0.01 < 0.05$

Следовательно, различие между группами статистически значимое для $\alpha = 0.05$.